### ISSUE TITLE
Add option to allow Lint/UselessAssignment to ignore when used with keyword args

### ISSUE THREAD

**Issue by rmm5t** [rubocop/rubocop#14976]
## Is your feature request related to a problem? Please describe.

I'm frustrated when rubocop presents a `Lint/UselessAssignment` offense when a variable is set and used as a keyword arg.

e.g. 

```ruby
status = @widget.available? ? :ok : :not_found
render json: @widget, status:
```

This results in:

```
app/controllers/widgets_controller.rb:7:5: C: [Correctable] Lint/UselessAssignment: Useless assignment to variable - status.
    status = @widget.available? ? :ok : :not_found
    ^^^^^^
app/controllers/widgets_controller.rb:8:5: C: [Correctable] Style/MultilineIfModifier: Favor a normal unless-statement over a modifier clause in a multiline statement.
    render json: @widget, status: ...
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

## Describe the solution you'd like

Instead, I'd like an option to ignore both of these cops when the variable assignment is used in a keyword arg. The concise keyword arg syntax is soetimes more readable than turning these two lines into one (with embedded ternary). It's also possible that I'd like to use other keyword arg variables like `flash:`, etc.

## Describe alternatives you've considered

None other than inline ternary, but this is less readable, especially if more than one conditional keyword arg is necessary:

```ruby
render json: @widget, status: (@widget.available? ? :ok : :not_found)
```

Disabling this cop isn't great, because it's beneficial in other usage scenarios.
